util/common/metrics.go

```diff
@@ -27,14 +38,71 @@ func StartSendingMetrics(receiver string, duration, sendingInterval time.Duratio
 			err = SendCollectDMetrics(metricPerInterval, sendingInterval, duration)
 		case "emf":
 			err = SendEMFMetrics(metricPerInterval, metricLogGroup, metricNamespace, sendingInterval, duration)
+		case "app_signals":
+			err = SendAppSignalMetrics() //does app signals have dimension for metric?
+		case "traces":
+			err = SendAppTraceMetrics(duration) //does app signals have dimension for metric?
+
 		default:
 		}
-
 	}()
 
 	return err
 }
 
+func SendAppTraceMetrics(duration time.Duration) error {
+	baseDir := getBaseDir()
+
+	for i := 0; i < int(duration/(5*time.Second)); i++ {
+		startTime := time.Now().UnixNano()
+		traceID := generateTraceID()
+		traceIDStr := hex.EncodeToString(traceID[:])
+
+		err := processTraceFile(filepath.Join(baseDir, "traces.json"), startTime, traceIDStr)
+		if err != nil {
+			fmt.Println("Error processing trace file:", err)
+			return err
+		}
+
+		time.Sleep(5 * time.Second)
+	}
+
+	return nil
+}
+
+func getBaseDir() string {
+	if runtime.GOOS == "windows" {
+		return "C:\\Users\\Administrator\\amazon-cloudwatch-agent-test\\test\\app_signals\\resources\\traces"
+	}
+	return "/Users/ec2-user/amazon-cloudwatch-agent-test/test/app_signals/resources/traces"
+}
+
+func generateTraceID() [16]byte {
+	var r [16]byte
+	epochNow := time.Now().Unix()
+	binary.BigEndian.PutUint32(r[0:4], uint32(epochNow))
+	rand.Read(r[4:])
+	return r
+}
+
+func processTraceFile(filePath string, startTime int64, traceID string) error {
+	data, err := os.ReadFile(filePath)
+	if err != nil {
+		return err
+	}
+
+	modifiedData := strings.ReplaceAll(string(data), "START_TIME", fmt.Sprintf("%d", startTime))
+	modifiedData = strings.ReplaceAll(modifiedData, "TRACE_ID", traceID)
+
+	url := "http://127.0.0.1:4316/v1/traces"
```

Comment:
nit: might want to make the port a variable for later use if we want to change it